PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/ui/webapp/deploysetting/WebAppSettingPanel.java

```diff
@@ -444,56 +487,79 @@ public void actionPerformed(AnActionEvent anActionEvent) {
         pnlWebAppTable = tableToolbarDecorator.createPanel();
     }
 
-    private void fillWebContainers() {
-        DefaultComboBoxModel<WebAppUtils.WebContainerMod> cbModel = new DefaultComboBoxModel<>();
-        for (WebAppUtils.WebContainerMod wc : WebAppUtils.WebContainerMod.values()) {
-            cbModel.addElement(wc);
+    @Override
+    public void fillSubscription(List<Subscription> subscriptions) {
+        for (Subscription subscription: subscriptions) {
+            cbSubscription.addItem(subscription);
+            if (Comparing.equal(subscription.subscriptionId(), defaultSubscription)) {
+                cbSubscription.setSelectedItem(subscription);
+            }
         }
-        cbWebContainer.setModel(cbModel);
+        defaultSubscription = null;
     }
 
-    private void fillThirdParty() {
-        DefaultComboBoxModel<AzulZuluModel> cbModel = new DefaultComboBoxModel<>();
-        for (AzulZuluModel jdk : AzulZuluModel.values()) {
-            if (jdk.isDeprecated()) {
-                continue;
+    @Override
+    public void fillResourceGroup(List<ResourceGroup> resourceGroups) {
+        for (ResourceGroup group: resourceGroups) {
+            cbExistResGrp.addItem(group);
+            if (Comparing.equal(group.name(), defaultResGrp)) {
+                cbExistResGrp.setSelectedItem(group);
             }
-            cbModel.addElement(jdk);
         }
-        cbThirdPartyJdk.setModel(cbModel);
+        defaultResGrp = null;
     }
 
-    private void fillPricingTier() {
-        for (Field field: PricingTier.class.getDeclaredFields()) {
-            int modifier = field.getModifiers();
-            if (Modifier.isPublic(modifier) && Modifier.isStatic(modifier) && Modifier.isFinal(modifier)) {
-                try {
-                    PricingTier pt = (PricingTier) field.get(null);
-                    cbPricing.addItem(pt);
-                } catch (IllegalAccessException e) {
-                    e.printStackTrace();
-                }
+    @Override
+    public void fillAppServicePlan(List<AppServicePlan> appServicePlans) {
+        for (AppServicePlan plan: appServicePlans) {
+            cbExistAppServicePlan.addItem(plan);
+            if (Comparing.equal(plan.id(), defaultServicePlan)) {
+                cbExistAppServicePlan.setSelectedItem(plan);
             }
         }
     }
 
     @Override
-    public void fillSubscription(List<Subscription> subscriptions) {
-        subscriptions.forEach(cbSubscription::addItem);
+    public void fillLocation(List<Location> locations) {
```

Comment:
cannot make sure about the sequence that fillXX will be called before resetfrom. So pass configuration into this class